src/LogFileReader.cpp

```diff
@@ -67,19 +72,27 @@ bool LogFileReader::readAll(const QString& fileName)
             emit batteryDataReceived(batteryData);
         }
     }
-
     return true;
 }
 
 // File input is a csv file in the format of hh:mm:ss:zzz, voltage, current
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
-    // TODO implement this first
-
-    // This is here to the compiler happy. Otherwise the compile
-    // will have an error warning about an unused variable. Remove this
-    // when you use it.
-    Q_UNUSED(line);
-    Q_UNUSED(batteryData);
-    return true;
+    QStringList lines = line.split(COMMA);
+    QTime time = QTime::fromString(lines[0], TIMEFORMAT);
+    bool notok = false;
+    if (time.isValid() && lines.count() == 3 && (lines[1].toDouble(&notok) != notok) && (lines[2].toDouble(&notok) != notok))
```

Comment:
I've made a comment on that but you probably missed it. 

lines[1].toDouble(&notok) would convert that line[1] into a double, and if the operation fail (not a number) it would convert line[1] into 0 with a boolean value of notok. That's why in that conditional sentence, i'm setting if there are not booleans values which mean the convert was successful then update the data